packages/flutter_web_plugins/lib/src/plugin_event_channel.dart

```diff
@@ -43,28 +60,52 @@ class PluginEventChannel<T> {
 
   /// The messenger used by this channel to send platform messages.
   ///
-  /// This must not be null. If not provided, defaults to
-  /// [pluginBinaryMessenger], which sends messages from the platform-side
-  /// to the framework-side.
-  BinaryMessenger get binaryMessenger =>
-      _binaryMessenger ?? pluginBinaryMessenger;
-  final BinaryMessenger _binaryMessenger;
-
-  /// Set the stream controller for this event channel.
+  /// When this is null, the [pluginBinaryMessenger] is used instead,
+  /// which sends messages from the platform-side to the
+  /// framework-side.
+  final BinaryMessenger binaryMessenger;
+
+  /// Use [setController] instead.
+  ///
+  /// This setter is deprecated because it has no corresponding getter,
+  /// and providing a getter would require making this class non-const.
+  @Deprecated(
+    'Replace calls to the "controller" setter with calls to the "setController" method. '
+    'This feature was deprecated after v1.23.0-7.0.pre.'
+  )
```

Comment:
Would be good to clean this up before stable release. How widespread is the usage of this setter?

Reply:
@Piinks is working on a deprecation policy.

Reply:
I'll make sure to send you the proposal, it will include what usage data I've been able to collect and some guidelines for cleaning up old code. <=1 week out from publishing :)